ui-tests/src/main/java/io/jenkins/plugins/analysis/warnings/AnalysisResult.java

```diff
@@ -0,0 +1,223 @@
+
+        return Tab.valueWithHref(extractRelativeUrl(activeTab.getAttribute("href")));
+    }
+
+    /**
+     * Returns the list of available tabs. These tabs depend on the available properites of the set of shown issues.
+     *
+     * @return the available tabs
+     */
+    public Collection<Tab> getAvailableTabs() {
+        return all(By.xpath("//a[@role='tab']")).stream()
+                .map(tab -> tab.getAttribute("href"))
+                .map(this::extractRelativeUrl)
+                .map(Tab::valueWithHref)
+                .collect(Collectors.toList());
+    }
+
+    private String extractRelativeUrl(final String absoluteUrl) {
+        return "#" + StringUtils.substringAfterLast(absoluteUrl, "#");
+    }
+
+    /**
+     * Returns the total number of issues. This method requires that one of the tabs is shown that shows the total
+     * number of issues in the footer. I.e. the {@link Tab#ISSUES} and {@link Tab#BLAMES}.
+     */
+    public int getTotal() {
+        String total = find(By.tagName("tfoot")).getText();
+
+        return Integer.parseInt(StringUtils.substringAfter(total, "Total "));
+    }
+
+    /**
+     * Returns the type of the rows in the issues table. Currently, code duplications have a different representation
+     * than all other static analysis tools.
+     *
+     * @return the row type
+     */
+    private IssuesTableRowType getIssuesTableType() {
+        if (ArrayUtils.contains(DRY_TOOLS, id)) {
+            return IssuesTableRowType.DRY;
+        }
+        return IssuesTableRowType.DEFAULT;
+    }
+
+    /**
+     * Opens the analysis details page and selects the specified tab.
+     *
+     * @param tab
+     *         the tab that should be selected
+     */
+    public void openTab(final Tab tab) {
+        open();
+
+        WebElement tabElement = getElement(By.id("tab-details")).findElement(tab.getXpath());
+        tabElement.click();
+    }
+
+    /**
+     * Opens the analysis details page, selects the tab {@link Tab#ISSUES} and returns the {@link PageObject} of the
+     * issues table.
+     *
+     * @return page object of the issues table.
+     */
+    public IssuesTable openIssuesTable() {
+        openTab(Tab.ISSUES);
+
+        WebElement issuesTab = find(By.id("issuesContent"));
+        return new IssuesTable(issuesTab, this, getIssuesTableType());
+    }
+
+    /**
+     * Opens a link on the page leading to another page.
+     *
+     * @param element
+     *         the WebElement representing the link to be clicked
+     * @param type
+     *         the class of the PageObject which represents the page to which the link leads to
+     *
+     * @return the instance of the PageObject to which the link leads to
+     */
+    // FIXME: IssuesTable should not depend on AnalysisResult
+    public <T extends PageObject> T openLinkOnSite(final WebElement element, final Class<T> type) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected @param tag for '<T>'.](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=5151082)